addons/io_hubs_addon/components/definitions/morph_audio_feedback.py

```diff
@@ -0,0 +1,74 @@
+import atexit
+import bpy
+from bpy.props import FloatProperty, StringProperty, EnumProperty
+from ..hubs_component import HubsComponent
+from ..types import Category, PanelType, NodeType
+
+
+def get_shape_keys(self, context):
+    shape_keys = []
+    if context.object.data.shape_keys:
+        for item in context.object.data.shape_keys.key_blocks:
+            shape_keys.append((item.name, item.name, ""))
+    return shape_keys
+
+
+class MorphAudioFeedback(HubsComponent):
+    _definition = {
+        'name': 'morph-audio-feedback',
+        'display_name': 'Morph Audio Feedback',
+        'category': Category.AVATAR,
+        'node_type': NodeType.NODE,
+        'panel_type': [PanelType.OBJECT],
+        'icon': 'MOD_SMOOTH'
+    }
+
+    name: StringProperty(
+        name="Name",
+        description="Name"
+    )
+
+    shape_key: EnumProperty(
+        name="Shape Key",
+        description="Shape key to morph",
+        items=get_shape_keys
+    )
+
+    minValue: FloatProperty(name="Min Value",
+                            description="Min Value",
+                            default=0.0,)
+
+    maxValue: FloatProperty(name="Max Value",
+                            description="Max Value",
+                            default=1.0)
+
+    @classmethod
+    def poll(cls, context):
+        return context.object.type == 'MESH'
+
+    @classmethod
+    def migrate(cls, version):
+        if version < (0, 1, 0):
+            for ob in bpy.data.objects:
+                if cls.get_name() in ob.hubs_component_list.items:
+                    ob.hubs_component_morph_audio_feedback.shape_key = (
+                        ob.hubs_component_morph_audio_feedback.name)
+
+    def draw(self, context, layout):
+        shape_keys = context.object.data.shape_keys
+        if shape_keys and len(shape_keys.key_blocks) > 0:
+            layout.prop(data=self, property="shape_key")
+            layout.prop(data=self, property="minValue")
+            layout.prop(data=self, property="maxValue")
+        else:
+            col = layout.column()
+            col.alert = True
+            col.label(text='No shape keys available',
+                      icon='ERROR')
+
+    def gather(self, export_settings, object):
+        return {
+            'name': object.hubs_component_morph_audio_feedback.shape_key,
+            'minValue': object.hubs_component_morph_audio_feedback.minValue,
+            'maxValue': object.hubs_component_morph_audio_feedback.maxValue
+        }
```

Comment:
I've managed to fix my two known issues, and now my code seems to be pretty solid (I have so far not been able to cause it to break), the only small thing I've noticed is that there is a lag if you have both the component and the list of shape keys visible at the same time and you rename the chosen shape key (this won't trigger an update, so the user will need to mouse over the component, or do something that does trigger an update to see the new value.  The same thing happens with the Loop Animation component).

This is kind of ugly, but it does seem to provide a decent UX, although I don't know how easy it will be to migrate/modify in the future, if desired.  Anyway, see what you think.

```suggestion
import atexit
import bpy
from bpy.props import IntProperty, FloatProperty, StringProperty, EnumProperty, PointerProperty
from ..hubs_component import HubsComponent
from ..types import Category, PanelType, NodeType

BLANK_SHAPE_KEY_IDENTIFIER = '56489845649846498select_a_shape_key8498416549846849849843'
shape_keys = []
shape_keys_change = 0


def get_shape_keys(self, context):
    global BLANK_SHAPE_KEY_IDENTIFIER
    global shape_keys
    global shape_keys_change

    change = len(shape_keys)
    shape_keys = []
    ob_shape_keys = self.ob.data.shape_keys
    index = 0

    shape_keys.append((BLANK_SHAPE_KEY_IDENTIFIER, 'Select a Shape Key',
                                  "Select a Shape Key", 'BLANK', index))

    index += 1

    if (self.name and not ob_shape_keys) or ((self.name and ob_shape_keys) and self.name not in ob_shape_keys.key_blocks):
        shape_keys.append((self.name, self.name,
                          "No matching shape key found", 'ERROR', index))
        index += 1

    if ob_shape_keys:
        for item in ob_shape_keys.key_blocks:
            if ' ' in item.name or ',' in item.name:
                shape_keys.append((item.name, item.name,
                                  "Shape key names can't have spaces or commas", 'ERROR', index))
                index += 1
            elif item == item.relative_key:
                pass
            else:
                shape_keys.append(
                    (item.name, item.name, "", 'SHAPEKEY_DATA', index))
                index += 1

    change = (change - len(shape_keys)) * -1
    if not shape_keys_change:
        shape_keys_change = change

    return shape_keys


def get_shape_key(self):
    global BLANK_SHAPE_KEY_IDENTIFIER
    global shape_keys
    global shape_keys_change

    if shape_keys_change:
        if self.active_shape_key_index > (len(shape_keys) -1) or shape_keys[self.active_shape_key_index][0] != self.active_shape_key_name:
            shape_key_names = [item[0] for item in shape_keys]
            if self.active_shape_key_name in shape_key_names:
                self.active_shape_key_index = shape_key_names.index(self.active_shape_key_name)
            else:
                self.active_shape_key_index = 0
                self.active_shape_key_name = BLANK_SHAPE_KEY_IDENTIFIER

        shape_keys_change = 0

    else:
        self.active_shape_key_name = shape_keys[self.active_shape_key_index][0]



    return self.active_shape_key_index


def set_shape_key(self, value):
    global shape_keys

    if self.name and self.name != shape_keys[value][0] and value != 0:
        self.name = ''

    self.active_shape_key_index = value
    self.active_shape_key_name = shape_keys[value][0]


class MorphAudioFeedback(HubsComponent):
    _definition = {
        'name': 'morph-audio-feedback',
        'display_name': 'Morph Audio Feedback',
        'category': Category.AVATAR,
        'node_type': NodeType.NODE,
        'panel_type': [PanelType.OBJECT],
        'icon': 'MOD_SMOOTH'
    }

    name: StringProperty(
        name="Name",
        description="Name"
    )

    active_shape_key_index: IntProperty(default=0, options={'HIDDEN'})

    active_shape_key_name: StringProperty(default='', options={'HIDDEN'})

    shape_key: EnumProperty(
        name="Shape Key",
        description="Shape key to morph",
        items=get_shape_keys,
        get=get_shape_key,
        set=set_shape_key
    )

    minValue: FloatProperty(name="Min Value",
                            description="Min Value",
                            default=0.0,)

    maxValue: FloatProperty(name="Max Value",
                            description="Max Value",
                            default=1.0)

    ob: PointerProperty(type=bpy.types.Object, options={'HIDDEN'})

    @classmethod
    def init(cls, obj):
        obj.hubs_component_morph_audio_feedback.ob = obj

    @classmethod
    def poll(cls, context):
        return context.object.type == 'MESH'

    @classmethod
    def migrate(cls, version):
        if version < (1, 0, 0):
            for ob in bpy.data.objects:
                if cls.get_name() in ob.hubs_component_list.items:
                    component = ob.hubs_component_morph_audio_feedback
                    component.ob = ob
                    if component.name in [item[0] for item in get_shape_keys(component, bpy.context) if item[3] != 'ERROR']:
                        component.shape_key = (component.name)
                        component.name = ''
                    else:
                        component.shape_key = (component.name)

    def draw(self, context, layout):
        global BLANK_SHAPE_KEY_IDENTIFIER

        if len(get_shape_keys(self, context)) > 1:
            layout.prop(data=self, property="shape_key")

            ob_shape_keys = context.object.data.shape_keys
            if (self.shape_key != BLANK_SHAPE_KEY_IDENTIFIER) and ((self.shape_key and not ob_shape_keys) or (ob_shape_keys and self.shape_key not in ob_shape_keys.key_blocks)):
                col = layout.column()
                col.alert = True
                col.label(text="No matching shape key found",
                          icon='ERROR')
            elif ' ' in self.shape_key or ',' in self.shape_key:
                col = layout.column()
                col.alert = True
                col.label(text="Shape key names can't have spaces or commas",
                          icon='ERROR')
            layout.prop(data=self, property="minValue")
            layout.prop(data=self, property="maxValue")
        else:
            col = layout.column()
            col.alert = True
            col.label(text='No shape keys available',
                      icon='ERROR')

    def gather(self, export_settings, object):
        global BLANK_SHAPE_KEY_IDENTIFIER

        component = object.hubs_component_morph_audio_feedback
        name = component.shape_key if component.shape_key != BLANK_SHAPE_KEY_IDENTIFIER else ''
        minValue = component.minValue
        maxValue = component.maxValue

        return {
            'name': name,
            'minValue': minValue,
            'maxValue': maxValue
        }

```